### DISCUSSION TITLE
Automating fund allocation

### DISCUSSION THREAD

**Discussion by MarkSFrancis** [community/community#115909] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I'd like to sponsor all the projects that I use on GitHub, but in the process of setting this up, I've discovered a few pain points for me that I'd like to highlight here:
- The setup is a challenge. The detection of projects via the bulk sponsorship is a great step, but I cannot simply press "please sponsor all these projects on my behalf". Instead, I must comb through them manually anyway.
- It's almost impossible for me to distribute my sponsorship fairly with my current budget. Because of minimum amounts for certain beneficiaries, I cannot afford to sponsor the maintainers of all the projects I am using. This effectively forces me to somehow choose which projects to sponsor and to cut the rest, even though I'm using them all. This creates a bias towards maintainers who maintain lots of projects, but that's not a great incentive for beneficiaries IMO. I'd like to see a few great projects maintained by a person rather than them having to stretch themselves thinly in order to get the support they deserve
- I have to manually reconfigure sponsorships every month to make sure that I'm sponsoring projects that I'm actually using

Ideally, I'd like to set some fixed monthly amount, and then have it automatically allocated across any maintainers of packages that are depended on equally, according to how many projects I have which depend on something they maintain. For example, if I depend on 500 projects, of which 100 are maintained by GitHub accounts with sponsorship programs, and a maintainer maintains 5 of those 100 projects, I'd like that maintainer to receive 5% of the funds. 

The main differences between this approach and the way that the Sponsorship program works today is that:
- It'd be really easy for me to make sure that the maintainers of projects I'm using are being supported fairly from my monthly donations, according to how much I'm using the dependencies that they maintain
- It'd be far easier for me to set up and maintain. Essentially from my perspective, I just say "I have $X that I want to contribute each month" and GitHub then figures out how to distribute that amount as fairly as it can on my behalf
- It takes the strain off the sponsor for having to decide which projects they do or don't sponsor, especially for projects that they might not even be aware they're using (such as dependencies of their dependencies)
- Each individual beneficiary would likely get a small amount from me, probably less than the current "minimum amount" that some sponsors have set. Unfortunately I cannot afford a monthly payment of $48,600, which would be the amount required to sponsor fairly and meet the minimum amounts set by the beneficiaries
- I'd like to believe that individuals would be less likely to have a commanding sponsorship for beneficiaries, as instead the beneficiaries would receive smaller amounts from a broader set of people. This reduces the risk of individuals deciding to withdraw funds having a major impact on beneficiaries

I believe that this set up would make it easier for a GitHub user to start to sponsor their dependencies, as the barrier to entry in terms of complexity of setup and maintenance would be lowered.

**Comment by anjuan** [comment]
Hi @MarkSFrancis, I'm the Engineering Manager for GitHub Sponsors. Thank you so much for sharing your thoughts about how to improve Sponsors. I'll take these ideas back to the team!

**Reply by nikeee** [reply]
I think what OP describes is essentially https://thanks.dev.

I just discovered https://github.com/sponsors/explore and all it needs is a "distribute X$ per month to these repositories". Basically [flattr](https://en.wikipedia.org/wiki/Flattr) that automatically distributes by project depdendencies. GitHub already seems to have the data needed to do this and this would be great (also to attract foss devs to GH I guess).

**Comment by hkme2008** [comment]
> ### Select Topic Area
> 
> Product Feedback
> 
> ### Body
> 
> I'd like to sponsor all the projects that I use on GitHub, but in the process of setting this up, I've discovered a few pain points for me that I'd like to highlight here:
> - The setup is a challenge. The detection of projects via the bulk sponsorship is a great step, but I cannot simply press "please sponsor all these projects on my behalf". Instead, I must comb through them manually anyway.
> - It's almost impossible for me to distribute my sponsorship fairly with my current budget. Because of minimum amounts for certain beneficiaries, I cannot afford to sponsor the maintainers of all the projects I am using. This effectively forces me to somehow choose which projects to sponsor and to cut the rest, even though I'm using them all. This creates a bias towards maintainers who maintain lots of projects, but that's not a great incentive for beneficiaries IMO. I'd like to see a few great projects maintained by a person rather than them having to stretch themselves thinly in order to get the support they deserve
> - I have to manually reconfigure sponsorships every month to make sure that I'm sponsoring projects that I'm actually using
> 
> Ideally, I'd like to set some fixed monthly amount, and then have it automatically allocated across any maintainers of packages that are depended on equally, according to how many projects I have which depend on something they maintain. For example, if I depend on 500 projects, of which 100 are maintained by GitHub accounts with sponsorship programs, and a maintainer maintains 5 of those 100 projects, I'd like that maintainer to receive 5% of the funds. 
> 
> The main differences between this approach and the way that the Sponsorship program works today is that:
> - It'd be really easy for me to make sure that the maintainers of projects I'm using are being supported fairly from my monthly donations, according to how much I'm using the dependencies that they maintain
> - It'd be far easier for me to set up and maintain. Essentially from my perspective, I just say "I have $X that I want to contribute each month" and GitHub then figures out how to distribute that amount as fairly as it can on my behalf
> - It takes the strain off the sponsor for having to decide which projects they do or don't sponsor, especially for projects that they might not even be aware they're using (such as dependencies of their dependencies)
> - Each individual beneficiary would likely get a small amount from me, probably less than the current "minimum amount" that some sponsors have set. Unfortunately I cannot afford a monthly payment of $48,600, which would be the amount required to sponsor fairly and meet the minimum amounts set by the beneficiaries
> - I'd like to believe that individuals would be less likely to have a commanding sponsorship for beneficiaries, as instead the beneficiaries would receive smaller amounts from a broader set of people. This reduces the risk of individuals deciding to withdraw funds having a major impact on beneficiaries
> 
> I believe that this set up would make it easier for a GitHub user to start to sponsor their dependencies, as the barrier to entry in terms of complexity of setup and maintenance would be lowered.
